### DISCUSSION TITLE
Inertia working weird, things that spin are not transfering properly, far too much spinning for heavy objects

### DISCUSSION THREAD

**Discussion by GeminiSquishGames** [appsinacup/godot-rapier-physics#267] [Q&A] [answered]
It seems like inertia isn't being calculated automatically. In default godot physics large colliding objects with smaller (less mass) ones work properly, however, everything acts like it has the same angular mass, like styrofoam or none at all, spinning wildly and often times asteroids gain angular velocity from little collision contact when they are actually metric tons in linear energy. I'm not sure how to make these objects act properly as in my situation it is a space game and things can't really have dampening, nor can I get the proper inertial effect from dampening. So I'm kinda stuck at this point or may just have to use default physics, which has it's own problems. Any insight would be a great help.

**Comment by Ughuuu** [comment]
Can you describe again in more detail? Firstly, you said in Godot Physics it works correctly, and in Rapier inertia isn't calculated automatically (it is calculated automatically).
Say exactly what you use, not asteroid, but be very specific, making sure to say exactly what collision shape and body type you use (eg convex polygon, rigidbody, etc).
As for dampening, I think you mean damping. And you shoudl be able to set damping to 0.
Make sure to explain better the use case you have, and try to write things separated, not all in 1 big phrase. Eg:

"""
I have a ship which is a rigidbody with shape circle..
Then the ship shoots particles that are..
The ship interacts with asteroids which are..

The ship doesnt rotate as much as it used to in Godot Physics. I assume the inertia isn't calculated automatically...
"""

Btw, you can get the computed inertia of the body (even if its set to automatic) by doing: ` PhysicsServer2D.body_get_param(body_rid, PhysicsServer2D.BODY_PARAM_INERTIA)` in order to compare inertia values between godot and rapier.

**Comment by GeminiSquishGames** [comment]
Ah, right I gotcha, I'm using a RigidBody2D node. It has a Polygon2D node for its visual, the points are generated so the shape is not the same as other polygon2D. 

This shape array is then given to a CollisionPolygon2D so that its visual shape is equivalent to its collision. This is used as the asteroid. 

I'm also using a small 8-point CollisionPolygon to cut holes in the main Polyshape/roid but only when it's hit with a bullet. 
This does mean recalculating the mass based on the area, but the mass checks out for where I want it to be per unit of area using a custom constant. 
This shouldn't have an effect and works fine in default and generally in Rapier2D when you don't take rotation into account. It may have worked at some time but I can't figure out what is causing the effect I will mention here.

I'm using a RigidBody 2D with a CircleShape2D with CollisionShape2D for my ship. I
t is controlled with apply_force that takes the angle of the RigidBody which is controlled using the mouse cursors angle to the RB2D and apply_torque to turn.

In the Default physics engine, the poly asteroids colliding will apply accurate linear and angular forces and move smoothly rotationally speaking, again no damping is used for angular. 

I don't have to set inertia, but if I do set it to say, mass,  I can get similar effects to how it works without setting inertia in Rapier2D. 

In Rapier with the same settings, no inertia set so should be automatic, it seems as though the angular movement is way too loose. 

The mass on my player object is set to 1000kg, or a metric ton, whereas some asteroid game objects are much heavier, say 80,000kg. 

The ship can easily stop a spinning asteroid and won't move at all, bouncing the asteroid rotationally back and forth. 

When asteroids collide they also exhibit this behavior, the linear momentum seems to spread correctly but the angular or rotation bounces erratically until it wears out, sometimes making them spin way faster than the frame rate can interpret it, which is way faster than I'd assume them to appear in space... at least my space, be it ever so humble.

**Comment by Ughuuu** [comment]
So the issue is that when you call apply_force and/or apply_torque it applies more rotation than in regular Godot Engine? You should open an issue instead and put a reproducible project.

**Comment by GeminiSquishGames** [comment]
No, only when bodies collide, applying forces is fine. If I can reproduce the problem I will do that, but so far I cannot get a simple example to exhibit the weirdness. That's why I didn't really want to make it an issue since it's probably something I'm doing weird since I'm used to the default physics engine. I'll play with things in my test project and see if I can reproduce faithfully.

**Comment by GeminiSquishGames** [comment, marked as answer]
I did have a working example of the problem, until I removed the Rapier extension and re-downloaded 0.8.6. Now things seem to work correctly, have no idea why exactly. I've done this reinstall a few times but it may have been something that didn't get updated on my end on or before 0.8.3.
